components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/DASRestClient.java

```diff
@@ -0,0 +1,256 @@
+
+        //parse the response back to the java objects
+        List<Result<T>> result = parse(response, type);
+
+        return result;
+    }
+
+    /**
+     * Top level method to make post request and return java object type response for first access time requests
+     *
+     * @param request FirstAccessRequestSearchBean representing the lucene json object for first access time search
+     * @param type    type of the expected java object type
+     * @param <T>     expected values of the Result object
+     * @return return list of Result objects containing the <T> values
+     * @throws JsonSyntaxException throws if error occur parsing response back to the java
+     * @throws IOException         throws if connection error occur to the REST API
+     */
+    public <T> List<Result<T>> doPost(FirstAccessRequestSearchBean request, Type type)
+            throws JsonSyntaxException, IOException {
+        //get the json string of the request object
+        String json = gson.toJson(request);
+
+        //doing a post request on the Search REST API
+        CloseableHttpResponse response = post(json, dasUrl + APIUsageStatisticsClientConstants.DAS_SEARCH_REST_API_URL);
+
+        //parse the response back to the java objects
+        List<Result<T>> result = parse(response, type);
+
+        return result;
+    }
+
+    /**
+     * use to check provided Table is present in the DAS Data access layer
+     *
+     * @param name Table name
+     * @return TableExistResponseBean which contain the row response from the REST API
+     * @throws JsonSyntaxException throws if parsing error occur
+     * @throws IOException         throws if connection problem occur
+     */
+    public TableExistResponseBean isTableExist(String name) throws JsonSyntaxException, IOException {
+
+        //crete the http get request method to RETS API
+        HttpGet getRequest = new HttpGet(
+                dasUrl + APIUsageStatisticsClientConstants.DAS_TABLE_EXIST_REST_API_URL + "?table=" + name);
+
+        //get the encoded REST API credentials
+        String cred = RestClientUtil.encodeCredentials(this.user, this.pass);
+        getRequest.addHeader(APIUsageStatisticsClientConstants.HTTP_AUTH_HEADER_NAME,
+                APIUsageStatisticsClientConstants.HTTP_AUTH_HEADER_TYPE + ' ' + cred);
+        TableExistResponseBean obj = null;
+        BufferedReader reader = null;
+        try {
+            //get the response
+            HttpResponse response = httpClient.execute(getRequest);
+
+            //get the buffered reader
+            reader = new BufferedReader(new InputStreamReader(response.getEntity().getContent()));
+
+            //get the expected result type
+            Type type = new TypeToken<TableExistResponseBean>() {
+            }.getType();
+
+            //pass to java object
+            obj = gson.fromJson(reader, type);
+        } finally {
+            //            httpClient.getConnectionManager().shutdown();
```

Comment:
fixed in commit: https://github.com/ruks/carbon-apimgt/commit/8d7937aa4f4d41ca0441b339bead82a16d6b6b41